cmd/entrypoints/serve.go

```diff
@@ -36,6 +34,7 @@ var serveCmd = &cobra.Command{
 				logger.Panicf(ctx, "Failed to Start profiling and Metrics server. Error, %v", err)
 			}
 		}()
+		server.SetMetricKeys(cfg.ApplicationConfiguration().GetTopLevelConfig())
```

Comment:
I do not know if you can do this? I think its a onStartup Property

Reply:
it seems it can be run only once. With this now you will have to make sure no one ever invokes it. I like it though!

https://github.com/flyteorg/flytestdlib/blob/15cac7bec8470974050ed02343cda8191e5f14bc/promutils/labeled/keys.go#L26

Reply:
yeah exactly, it'll panic if we it's called >1x, but for now people shouldn't be calling it themselves except for weird cases like single binary, which i will update to set the labels via the config after this release